specification/ai/OpenAI.Assistants/messages/models.tsp

```diff
@@ -207,3 +239,199 @@ model MessageFile {
+  /** The type of content for this content part, which is always "text." */
+  type: "text";
+
+  /** The text content details. */
+  text?: MessageDeltaTextContent;
+}
+
+/** Represents the data of a streamed text content part within a streaming message delta chunk. */
+@added(ServiceApiVersions.v2024_02_15_preview)
+model MessageDeltaTextContent {
+  /** The data that makes up the text. */
+  value?: string;
+
+  /** Annotations for the text. */
+  annotations?: MessageDeltaTextAnnotation[];
+}
+
+/** The abstract base representation of a streamed text content part's text annotation. */
+@discriminator("type")
+@added(ServiceApiVersions.v2024_02_15_preview)
+model MessageDeltaTextAnnotation {
+  /** The index of the annotation within a text content part. */
+  index: int32;
+
+  /** The type of the text content annotation. */
+  type: string;
+}
+
+/** Represents a streamed file citation applied to a streaming text content part. */
+@added(ServiceApiVersions.v2024_02_15_preview)
+model MessageDeltaTextFileCitationAnnotationObject
+  extends MessageDeltaTextAnnotation {
+  /** The type of the text content annotation, which is always "file_citation." */
+  type: "file_citation";
+
+  /** The file citation information. */
+  @encodedName("application/json", "file_citation")
+  fileCitation?: MessageDeltaTextFileCitationAnnotation;
+
+  /** The text in the message content that needs to be replaced */
+  text?: string;
+
+  /** The start index of this annotation in the content text. */
+  @encodedName("application/json", "start_index")
+  startIndex?: int32;
+
+  /** The end index of this annotation in the content text. */
+  @encodedName("application/json", "end_index")
+  endIndex?: int32;
+}
+
+/** Represents the data of a streamed file citation as applied to a streaming text content part. */
+@added(ServiceApiVersions.v2024_02_15_preview)
+model MessageDeltaTextFileCitationAnnotation {
+  /** The ID of the specific file the citation is from. */
+  @encodedName("application/json", "file_id")
+  fileId?: string;
+
+  /** The specific quote in the cited file. */
+  quote?: string;
+}
+
+/** Represents a streamed file path annotation applied to a streaming text content part. */
+@added(ServiceApiVersions.v2024_02_15_preview)
+model MessageDeltaTextFilePathAnnotationObject
+  extends MessageDeltaTextAnnotation {
+  /** The type of the text content annotation, which is always "file_path." */
+  type: "file_path";
+
+  /** The file path information. */
+  @encodedName("application/json", "file_path")
+  filePath?: MessageDeltaTextFilePathAnnotation;
+
+  /** The start index of this annotation in the content text. */
+  @encodedName("application/json", "start_index")
+  startIndex?: int32;
+
+  /** The end index of this annotation in the content text. */
+  @encodedName("application/json", "end_index")
+  endIndex?: int32;
+
+  /** The text in the message content that needs to be replaced */
+  text?: string;
```

Comment:
Both for this variant extending `MessageDeltaTextAnnotation` and `MessageDeltaTextFileCitationAnnotationObject` I would love to get your take on the field `text` seems to be oddly repeated everywhere. Not sure what is the best way to approach it @trrwilson